test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
Well that makes it pretty easy then, can you `#derive(Default)` on the multiple Termios struct declarations and then use that in your tests. I'll then make sure those tests don't break when resolving #527. This is being tracked in rust-lang/libc#573.